tests/test_java.py

```diff
@@ -811,3 +813,316 @@ def flush_output_and_stop_async_profiler(self: AsyncProfiledProcess, *args: Any,
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    java_cli_flags: str,
+    java_env_flags: str,
+    expected_flags: List[Dict[str, Union[None, str, List[str]]]],
+    application_pid: int,
+    application_image_tag: str,
+    profiler_state: ProfilerState,
+) -> None:
+    """
+    1. Tests collections jvm flags from env & commandline origins and reporting the correct origin
+    2. Tests collecting only specific flags
+    """
+    with make_java_profiler(profiler_state, java_collect_jvm_flags="SelfDestructTimer,PrintCodeCache") as profiler:
+        # When running a container manually we can't use application_pid fixture as it will come from the fixture
+        # container and not from the manually started one
+        with _application_docker_container(
+            docker_client,
+            application_docker_image,
+            [],
+            [],
+            application_docker_command=[
+                "bash",
+                "-c",
+                f"export JAVA_TOOL_OPTIONS={java_env_flags}; exec java {java_cli_flags} -jar Fibonacci.jar",
+            ],
+        ) as container:
+            actual_flags = profiler._metadata.get_jvm_flags_serialized(psutil.Process(container.attrs["State"]["Pid"]))
+            assert_jvm_flags_equal(actual_jvm_flags=actual_flags, expected_jvm_flags=expected_flags)
+
+
+@pytest.mark.parametrize("java_cli_flags", ["-XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=95"])
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("expected_flags", [[]])
+def test_collect_flags_unsupported_filtered_out(
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    java_cli_flags: str,
+    expected_flags: List[Dict[str, Union[None, str, List[str]]]],
+    application_pid: int,
+    application_image_tag: str,
+    profiler_state: ProfilerState,
+) -> None:
+    """
+    Tests filtering of jvm flags we don't support collecting
+    """
+    with make_java_profiler(profiler_state, java_collect_jvm_flags="MaxHeapFreeRatio, MinHeapFreeRatio") as profiler:
+        # When running a container manually we can't use application_pid fixture as it will come from the fixture
+        # container and not from the manually started one
+        with _application_docker_container(
+            docker_client,
+            application_docker_image,
+            [],
+            [],
+            application_docker_command=[
+                "bash",
+                "-c",
+                f"exec java {java_cli_flags} -jar Fibonacci.jar",
+            ],
+        ) as container:
+            assert (
+                profiler._metadata.get_jvm_flags_serialized(psutil.Process(container.attrs["State"]["Pid"]))
+                == expected_flags
+            )
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("expected_flags", [None])
```

Comment:
UX wise let's return an empty list for `none`? Instead of `None`

Reply:
But its would be nice to tell apart when all flags were filtered vs when none were requested, wouldn't it?

Reply:
"No flags were requested" is not a legit use case. There's no reason to create multiple edge cases where you only need one. Empty list is no flags and that's it